ratis-netty/src/main/java/org/apache/ratis/netty/client/DataStreamReplyDecoder.java

```diff
@@ -29,25 +29,22 @@
 
   @Override
   protected void decode(ChannelHandlerContext channelHandlerContext,
-                        ByteBuf byteBuf, List<Object> list) throws Exception {
+                        ByteBuf byteBuf, List<Object> list) {
 
     if(byteBuf.readableBytes() >= 24){
```

Comment:
Ah I see. It seems that there are at least 3 long which counts for bytes in the buffer. That's why here checks >= 24 bytes.

It will be useful to have javadoc on encode/decode functions to document the encoding format.

Reply:
This may be a good suggestion but is not related to this issue.